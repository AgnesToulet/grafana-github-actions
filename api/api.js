"use strict";
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See LICENSE in the project root for license information.
 *--------------------------------------------------------------------------------------------*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.projectType = void 0;
var projectType;
(function (projectType) {
    projectType[projectType["Project"] = 0] = "Project";
    projectType[projectType["ProjectNext"] = 1] = "ProjectNext";
})(projectType = exports.projectType || (exports.projectType = {}));
//# sourceMappingURL=api.js.map