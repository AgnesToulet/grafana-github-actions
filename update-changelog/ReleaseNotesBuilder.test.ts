import dotenv from 'dotenv'
import { Query } from '../api/api'
import { OctoKit } from '../api/octokit'
import { Testbed, TestbedIssueConstructorArgs } from '../api/testbed'
import {
	ReleaseNotesBuilder,
	CHANGELOG_LABEL,
	GRAFANA_UI_LABEL,
	BREAKING_CHANGE_LABEL,
	BREAKING_SECTION_START,
	ENTERPRISE_LABEL,
} from './ReleaseNotesBuilder'

describe('ReleaseNotesBuilder', () => {
	it('Should build correct release notes', async () => {
		const issues: TestbedIssueConstructorArgs[] = [
			{
				issue: {
					title: 'Has no labels',
				},
			},
			{
				issue: {
					title: 'Alerting: Fixed really bad issue',
				},
				labels: [CHANGELOG_LABEL],
			},
			{
				issue: {
					title: 'Login: New feature',
				},
				labels: [CHANGELOG_LABEL],
			},
			{
				issue: {
					title: 'API: Fixed api issue',
					isPullRequest: true,
					author: {
						name: 'torkelo',
					},
				},
				labels: [CHANGELOG_LABEL],
			},
			{
				issue: {
					title: 'Button: Changed prop name for button',
					isPullRequest: true,
					author: {
						name: 'torkelo',
					},
				},
				labels: [CHANGELOG_LABEL, GRAFANA_UI_LABEL],
			},
			{
				issue: {
					title: 'Dashboard: Issue with deprecation notice',
					body: `
asdasd
asdasd

### ${BREAKING_SECTION_START}
Here is the content of this breaking change notice.`,
					isPullRequest: true,
					author: {
						name: 'torkelo',
					},
				},
				labels: [CHANGELOG_LABEL, BREAKING_CHANGE_LABEL],
			},
		]

		const queryRunner = async function* (
			query: Query,
		): AsyncIterableIterator<TestbedIssueConstructorArgs[]> {
			yield issues
		}

		const testbed = new Testbed({
			queryRunner,
			milestone: { closed_at: '2020-11-11T17:15:26Z' },
		})

		const builder = new ReleaseNotesBuilder(testbed, '7.3.3')
		const text = await builder.buildReleaseNotes({ useDocsHeader: false })
		expect(text).toMatchSnapshot()
	})

	it('Should be able to get notes with docs header', async () => {
		const issues: TestbedIssueConstructorArgs[] = [
			{
				issue: {
					title: 'Alerting: Fixed really bad issue',
				},
				labels: [CHANGELOG_LABEL],
			},
		]

		const queryRunner = async function* (
			query: Query,
		): AsyncIterableIterator<TestbedIssueConstructorArgs[]> {
			yield issues
		}

		const testbed = new Testbed({
			queryRunner,
			milestone: { closed_at: '2020-11-11T17:15:26Z' },
		})

		const builder = new ReleaseNotesBuilder(testbed, '7.3.3')
		const text = await builder.buildReleaseNotes({ useDocsHeader: true })
		expect(text).toMatchInlineSnapshot(`
		"# Release notes for Grafana 7.3.3

		### Bug fixes

		* **Alerting**: Fixed really bad issue. [#1](https://github.com/grafana/grafana/issues/1)
		* **Alerting**: Fixed really bad issue. (Enterprise)
		"
	`)

		expect(builder.getTitle()).toBe('Release notes for Grafana 7.3.3')
	})

	it.skip('integration test', async () => {
		dotenv.config()

		const token = process.env.TOKEN
		const repo = process.env.REPO
		const owner = process.env.OWNER

		const octokit = new OctoKit(token, { repo, owner })
		const builder = new ReleaseNotesBuilder(octokit)
		const text = await builder.buildReleaseNotes('7.3.2')
		console.log(text)
		expect(text).toEqual('asd')
	})
})
