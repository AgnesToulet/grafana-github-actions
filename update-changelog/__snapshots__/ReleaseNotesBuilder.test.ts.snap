// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseNotesBuilder Should build correct release notes 1`] = `
"### Features / Enhancements

* **Dashboard**: Issue with deprecation notice. [#1](https://github.com/grafana/grafana/pull/1), [@torkelo](https://github.com/torkelo)
* **Login**: New feature. [#1](https://github.com/grafana/grafana/issues/1)

### Bug Fixes

* **API**: Fixed api issue. [#1](https://github.com/grafana/grafana/pull/1), [@torkelo](https://github.com/torkelo)
* **Alerting**: Fixed really bad issue. [#1](https://github.com/grafana/grafana/issues/1)

### Breaking changes

Here is the content of this breaking change notice. Issue [#1](https://github.com/grafana/grafana/issues/1)

### Plugin development fixes & changes

* **Button**: Changed prop name for button. [#1](https://github.com/grafana/grafana/pull/1), [@torkelo](https://github.com/torkelo)
"
`;
